server/src/main/java/org/candlepin/pki/PKIUtility.java

```diff
@@ -41,48 +40,54 @@
 import java.security.cert.CertificateFactory;
 import java.security.cert.X509CRL;
 import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPrivateKey;
 import java.security.spec.InvalidKeySpecException;
 import java.security.spec.PKCS8EncodedKeySpec;
 import java.security.spec.X509EncodedKeySpec;
 import java.util.Date;
 import java.util.List;
 import java.util.Set;
 
+import javax.inject.Inject;
+
 /**
  * PKIUtility
  */
-public abstract class PKIUtility {
+public class PKIUtility {
```

Comment:
What value are we getting out of this class if it's just passthrough to the implemented interface? Seems like we could drop this entirely, inject the PKIProviderUtility as needed and skip this class entirely.

Reply:
PKIUtility implements crypto methods using the JCA interfaces (i.e. stuff that comes with the JDK like verifying digital signatures), but there's other stuff we need to do that requires us to dip into the classes of the crypto provider.  PKIUtility is meant to be the one-stop-shop and uses composition to access the provider specific implementations (PKIProviderUtility).  The goal is to give us the flexibility in the future to switch providers again if we need to.

I will agree that the architecture/naming is sub-optimal.  For this first pass through I was mainly focusing on getting JSS to work without destroying all the BouncyCastle stuff as well.  Let me know if you have some ideas about organization/architecture/etc.